migrate/sqlschema/inspector.go

```diff
@@ -0,0 +1,200 @@
+package sqlschema
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/uptrace/bun"
+	"github.com/uptrace/bun/schema"
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+type InspectorDialect interface {
+	schema.Dialect
+	Inspector(db *bun.DB, excludeTables ...string) Inspector
+
+	// EquivalentType returns true if col1 and co2 SQL types are equivalent,
+	// i.e. they might use dialect-specifc type aliases (SERIAL ~ SMALLINT)
+	// or specify the same VARCHAR length differently (VARCHAR(255) ~ VARCHAR).
+	EquivalentType(Column, Column) bool
+}
+
+// Inspector reads schema state.
+type Inspector interface {
+	Inspect(ctx context.Context, schemaName string) (Database, error)
```

Comment:
I'd say schema name should be passed when creation an inspector:

```
dialect.Inspector(db, sqlschema.WithSchemaName(...), sqlschema.ExcludeTables(...))
```

Maybe event rename `Inspector` to `NewInspector` for clarity.

Reply:
Addressed in #1058